Dashboard v2: Add `SummaryButton` component

ntsekouras opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Resolves: DOTCOM-10638

## Proposed Changes

`SummaryButton` is a component that is going to be reused in many places around the new hosting dashboard. This PR implements that.

## Notes
`SummaryButton` per the suggestions is under `@automattic/components`. I had to also add(duplicate) the `CoreBadge` from calypso, but I'm not exporting it.. Since the goal is to use `CoreBadge` as well in other designs, this is probably the way to go, and remove the instance from `calypso/components`. Unless I missed something and we can import that in `@automattic/components`.



## Tasks

- [x] Iterate on design for improvements
- [x] Responsive design (probably needs tweaks, especially with badges and 'medium/high' density
- [x] Add image in `decorator` in storybook
- [x] Write unit tests in the end



## Testing Instructions
1. in storybook `yarn components:storybook:start`
2. I've also updated the temp buttons in `me/billing` to use them as well.


### Screenshot with some variations
<img width="518" alt="Screenshot 2025-04-28 at 12 22 11 PM" src="https://github.com/user-attachments/assets/8df581ff-790a-4494-82a2-2cfb564740bc" />



## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

ntsekouras committed:
Dashboard v2: Add `SummaryButton` component

youknowriad commented on client/dashboard/summary-button/index.tsx:
```diff
@@ -0,0 +1,92 @@
+	return (
+		<Button
+			ref={ ref }
+			href={ to }
```
to is a route or an href?

ntsekouras replied:
It's [mapped](https://github.com/Automattic/wp-calypso/pull/102827/files#diff-a108a3f5a4f438d0de7037662428d380968315f702076c13b99c893757c3a22aR30) to the `href` prop.

If we want tanstack routing though we can use `RouterLinkSummaryButton` and pass a the `route.to` value.

I mention this in the PR's description and the reasoning for this.

youknowriad replied:
It should be named "href" for `SummaryButton` then because `SummaryButton` is just a regular button with some design choices, it has nothing to do with the router.

youknowriad reviewed (commented):
My understanding is that SummaryButton is supposed to be a generic UI component (so not connected to a router) and might live in automattic/components and RouterLinkSummaryButton is a wrapper to connect it to the router, right?

In that case, they should be in two different folders no?

youknowriad commented on client/dashboard/summary-button/types.ts:
```diff
@@ -0,0 +1,74 @@
+type Density = 'low' | 'medium' | 'high';
```
Is there really a need to separate the "types" of a single component props to a separate file?

youknowriad replied:
No strong opinion here, just asking.

ntsekouras replied:
There is no need of course. I just followed the same practice with core components and want to see later if we can auto generate the docs from it. It might be simpler with a separate file. It's minor though.

youknowriad commented on client/dashboard/summary-button/style.scss:
```diff
@@ -0,0 +1,65 @@
+.summary-button {
```
Feels like there's a lot of CSS, maybe our components don't compose well?

ntsekouras replied:
I wouldn't say much TBH.. We need the first lines to override the `Button` styles when we have an `a` tag and some colors on hover, etc..

Also regarding the `Text` usages and rules for `summary-button-title` for example, I think it's fine to add a line of css and get the benefits of `Text`.

youknowriad replied:
I think CSS within the reusable components is fine, so I'm not too concerned. I believe we should try to avoid CSS more at the app level but still something to pay attention to.

jameskoster commented:
Solid start! :) Some small design tweaks:

## Low density
* Padding should be `24px` on all sides. I appreciate `Card` doesn't provide this config 😓 Perhaps we add 8px vertical padding to the `HStack` container?
* Strapline should use `heading-small` [typography token](https://wordpress.github.io/gutenberg/?path=/docs/tokens-typography--page), with `text-transform: uppercase`.
* Gap between icon and other content should be `16px`.
* Gap between strapline/title/description container and badges should be `12px`.

## Medium density
* Title should be vertically aligned with icon / badges / chevrons.
* Title should use `body-medium` [typography token](https://wordpress.github.io/gutenberg/?path=/docs/tokens-typography--page).
* Should not be rounded; the medium variant is designed to be shown in a card as a list, with no gaps between.
* Should only have a bottom border.
* Gap between badges and chevron should be `12px`.

## High density
* Icon / title / badges / chevron should be vertically aligned.
* Title should use `body-medium` [typography token](https://wordpress.github.io/gutenberg/?path=/docs/tokens-typography--page).
* Should not have a border.
* Border radius should be $radius-small.

## Hover 
* Background should be theme color @ 2% opacity.
* Chevron and decorative icon should also be theme colored.
* Border color should change to theme color @ 8% opacity.

## Focus
* Focus ring radius should match the border of the button (`$radius-large` for low density, no radius for medium density, `$radius-small` for high density).

youknowriad commented:
Personally I'd love if we add the variations as we need them, rather than building all the variations and use them later. I don't want us to end up with components with 10 variations and we use only 2

jasmussen commented:
Indeed, what would it look like if we had only "Medium" density to start?

To be fair I do think we need the higher and lower, and that these will get vetted as we build out the sections. But if we start with just a single density and then add the others when they become necessary, which default should we start with?

jameskoster commented:
The host dash calls for the low and medium density variants. No strong feelings about whether they're implemented sequentially. 

The high density variant is mostly for things like the quick edit panel / inspector in core. So much lower prio.

ciampo commented on client/dashboard/summary-button/index.tsx:
```diff
@@ -0,0 +1,92 @@
+function SummaryButton(
+	{
+		title,
+		to,
```
Seeing that the ref type is `HTMLAnchorElement`, is the expectation that this component is always going to get a `to` prop? In other words, is this component always going to render an anchor link? 

In that case, calling it `button` may be confusing (cc @mirka who has been looking at standardising links in calypso)

ntsekouras replied:
No. I've updated to `React.ForwardedRef< HTMLAnchorElement | HTMLButtonElement >`. It might be a bit confusing, but it's a wrapper of `Button` component which does the same thing - might render either a `button` or an `a` tag.

ciampo commented on client/dashboard/summary-button/index.tsx:
```diff
@@ -0,0 +1,92 @@
+		strapline,
+		fields,
+		leadsToNestedPage = true,
+		onClick = noop,
```
NIt: I don't think we need a default `noop` value?

ciampo commented on client/dashboard/summary-button/index.tsx:
```diff
@@ -0,0 +1,92 @@
+	const hasLowDensity = density === 'low';
+	const badges = fields?.length && (
+		<HStack spacing={ 1 } justify="flex-start" style={ { minWidth: 'fit-content' } }>
+			{ fields?.map( ( field: SummaryButtonFieldProps ) => (
```
If `fields` is already typed in `SummaryButtonProps`, we shouldn't need to specify the `SummaryButtonFieldProps` prop explicitly (TS should already know)

ciampo commented on client/dashboard/summary-button/index.tsx:
```diff
@@ -0,0 +1,92 @@
+	ref: React.ForwardedRef< HTMLAnchorElement >
+) {
+	const hasLowDensity = density === 'low';
+	const badges = fields?.length && (
```
Nit, but `SummaryBadges` could be moved to a separate React component (even inside the same file)

ciampo commented on client/dashboard/summary-button/style.scss:
```diff
@@ -0,0 +1,65 @@
+		background-color: inherit;
+	}
+
+	&:hover:not(:disabled, [aria-disabled="true"], .is-pressed) {
+		background: color-mix(in srgb, var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9)) 4%, transparent);
+		color: var(--wp-components-color-accent-darker-20, var(--wp-admin-theme-color-darker-20, #183ad6));
+
+		.summary-button-title {
+			color: var(--wp-components-color-accent-darker-20, var(--wp-admin-theme-color-darker-20, #183ad6));
+		}
+	}
+
+	&:disabled, &[aria-disabled="true"] {
+		.summary-button-card-body {
+			opacity: 0.6;
+
+			.summary-button-title {
+				// TODO: can I get this from a var?
+				color: #757575;
+			}
+		}
+	}
```
The need to override so many `Button` styles (padding, opacity, color and bg color) of the `Button` component could be a sign that we need to improve its design / relationship with the DS:

- is the `Button` component not up to date with the latest `DS` specs?
- is the `SummaryButton` design is not using the DS button?

cc @jameskoster @matt-west

jasmussen replied:
To answer, by the way, 757575 is $gray-700

ntsekouras replied:
I've started to think that maybe we should have a custom `button|a` component and not the `Button` from core components. We'd still need some color styles for hover, etc.. and we'd need to have some code from core (mostly about `accessibleWhenDisabled` handling.

My rationale is around the opinionated `height` prop and lot's of functionality we don't need (icons, tooltips, shortcuts, etc..). 

I'll give it a go to update the PR.

jameskoster replied:
@ciampo I know Button is a bit of a 🌶️ topic. Could it make sense to have a less opinionated primitive Button component that handles basic stuff, then 'wrapper' components for specific use cases like primary buttons, summary buttons, etc. A bit like BaseControl. 🤷‍♂️

ntsekouras replied:
Actually I went in a rabbit hole trying that (with TS types mostly). The custom styles to match Button for various states plus the code for the 'thin' button, was way too much than adding just a few css rules..

So, I'll use `Button` for now.

ciampo replied:
Very hot topic, with many alternative paths that we could take — from keeping wp/components' `Button` untouched, to making overrides in a8c/components, to creating a separate `Button` that is a lot leaner and generic/reusable. (cc @mirka )

mirka replied:
Here's the [original Gutenberg issue](https://github.com/WordPress/gutenberg/issues/67320) with some prior discussion of an unstyled Button.

ciampo commented on client/dashboard/summary-button/summary-button.stories.tsx:
Nit: we usually call this file `index.stories.tsx` as it;s already in the `summary-button` folder

ciampo commented on client/dashboard/summary-button/summary-button.stories.tsx:
```diff
@@ -0,0 +1,96 @@
+			mapping: fieldOptions,
+			description: 'Pre-defined badge sets to display',
+		},
+		density: {
+			control: 'radio',
+			options: [ 'low', 'medium', 'high' ],
+		},
```
I don't think we need to specify an explicit Storybook control — it should be generated automatically from the `density` prop types

ntsekouras replied:
I'm not sure Storybook can create a `radio/select` control based on `type Density = 'low' | 'medium';`, as it's a string and would create a `string` control. Is there another way I'm not aware of?

mirka replied:
The component needs to have a named export for the Storybook docgen to pick up type information properly. Same as what's described [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#considerations-for-the-docgen) in the Gutenberg repo.

ntsekouras replied:
Cool, thank you!

ciampo commented on client/dashboard/summary-button/index.tsx:
```diff
@@ -0,0 +1,92 @@
+			disabled={ disabled }
+			accessibleWhenDisabled
+		>
+			<Card className="summary-button-card">
```
(_Kind of related to [this point](https://github.com/Automattic/wp-calypso/pull/102827/files#r2059586860)_)

In theory, a `div` is not an allowed child of the `<button />` element — but it is an allowed child of the `<a />` element. If this component is not guaranteed to always render an `<a />`, we need to rework its markup to make sure it produces valid HTML.

ntsekouras replied:
You're right! I'll see if can do all these with spans.

ciampo commented on client/dashboard/summary-button/style.scss:
```diff
@@ -0,0 +1,65 @@
+		}
+	}
+
+	.summary-button-strapline {
+		text-transform: uppercase;
+	}
```
The text component has an `upperCase` prop that could replace the need for these styles

ciampo commented on client/dashboard/summary-button/style.scss:
```diff
@@ -0,0 +1,65 @@
+		flex-shrink: 0;
+	}
+
+	.summary-button-decoration {
+		img {
+			// TODO: revisit styles with design.. These are placeholder styles.
+			width: 20px;
+			height: 20px;
+			flex-shrink: 0;
+		}
+	}
```
If the `SummaryButton` component allows its consumers to pass any `React.Element` for the `decoration`, I don't think that the styles for the decoration should be inside the `SummaryButton` component.

In other words — for better separation / encapsulation / reusability, it's fine to have styles for `.summary-button-decoration`, but the styles for the `.summary-button-decoration img` should be provided by the consumer of `SummaryButton`, eg

```
<SummaryButton decoration={ <img className="my-image" /> } />

// styles
.my-image {
 ...
}
```

ntsekouras replied:
I think we need to be consistent here and probably the way is to have two props, that would also simplify the API for consumers:
1. icon
2. decorationImage

What do you think @jameskoster? Would we need something different than an `image` or an `icon`?

jameskoster replied:
Images and icons are the only uses I saw in the dotcom host dash designs so seems like a good place to start.

What would happen if a consumer supplied both an icon and an image? In terms of display that's a situation we'd like to avoid.

ntsekouras replied:
>What would happen if a consumer supplied both an icon and an image?

That's a good question. Since there are no back-compat issues here, we can start by having the single `decoration` prop and revisit if needed based on usage.

ciampo commented on client/dashboard/summary-button/style.scss:
```diff
@@ -0,0 +1,65 @@
+.summary-button {
+	padding: 0;
+	height: inherit;
```
Is `height: inherit` necessary on the `SummaryButton` component itself? For better reusaibility of the component, it feels like something that should be applied by the consumer of `SummaryButton` when needed.

ntsekouras replied:
Button, has fixed height. I'm already removing `Card and CardBody` components because they render `divs` and will see if it makes sense to keep Button too, but for now it seems okay.

ciampo replied:
Re. `Card`, could you use the `as="span"` prop?

ciampo replied:
Re. `Button`, I see — maybe then `initial` or `default` rather than `inherit` is a better default?

ntsekouras replied:
>Re. Card, could you use the as="span" prop?

Yes, but there is a context provider that renders a `div`.

ciampo commented on client/dashboard/summary-button/style.scss:
```diff
@@ -0,0 +1,65 @@
+
+	.summary-button-navigation-icon,
+	.summary-button-decoration svg {
+		fill: #757575;
```
I wonder if there's a variable that we can use instead of a hardcoded color

ntsekouras replied:
Yes, I have the same comment above in a TODO.

ntsekouras commented:
@jameskoster I'm removing `Card and CardBody` components right now, because they render a `div` (context provider), that cannot be avoided and results in invalid HTML inside the `button` tag. Any objections to this or suggestions?

jameskoster commented:
Yes totally! I was thinking about suggesting that anyway that given the feedback around radius, spacing, etc :)

ntsekouras committed:
feedback part 1

jameskoster commented:
<!-- linear:replyTo:2790604995 -->

Well now you really got me thinking 😆 

I guess the question to answer is whether the similar appearance between `Card` and `SummaryButton` are intentional or incidental. If we updated the `Card` background color, would we expect the low density `SummaryButton` to remain aligned? In most cases I think that's probably a *yes*, so perhaps `Card > Button` would work. 

However, would it not be problematic that we'd want the `Button` states (e.g. Hover) to affect the `Card` styles (background, border-color)?

It also seems important to note that only the low density `SummaryButton` has visual resemblance with `Card`. The medium density version is explicitly designed to be placed inside a `Card`, e.g.:

<img src="https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/911599f3-9d26-44e5-b6c5-fbe86d9e8b0b/26b18498-47ef-4a6f-9780-9e1d74b05cd2?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi85MTE1OTlmMy05ZDI2LTQ0ZTUtYjZjNS1mYmU4NmQ5ZThiMGIvMjZiMTg0OTgtNDdlZi00YTZmLTk3ODAtOWUxZDc0YjA1Y2QyIiwiaWF0IjoxNzQ1NTk1MjgwLCJleHAiOjMzMzE2MTU1MjgwfQ.U5Y3_5gZgT-XSOkiEBT5o9Jb3r5UgVwSEeS1DHRwytc " alt="Screenshot 2025-04-25 at 16.33.12.png" width="795" data-linear-height="481" />

ciampo commented:
<!-- linear:replyTo:2790604995 -->

And I'll go one step further, since we're here — is `Card` still a relevant abstraction today? Or do you think it needs a few tweaks (or major changes) to fit today's DS direction?

ntsekouras committed:
move SummaryButton to `@automattic/components`

youknowriad commented on packages/components/src/summary-button/README.md:
```diff
@@ -0,0 +1,42 @@
+| strapline         | string                    | -       | A brief, optional line of text used to highlight important information, such as a warning or status.                                                                  |
+| decoration        | React.ReactElement        | -       | An optional visual element such as an icon or small illustration to enhance visual context or reinforce the category.                                                 |
+| fields            | SummaryButtonFieldProps[] | -       | This property is used to display `CoreBadge` instances per field. For this reason we need to define the props that match the `CoreBadge` component (intent and text). |
+| leadsToNestedPage | boolean                   | true    | A flag that indicates whether the button leads to a deeper level of navigation or a separate detail page. When `true` it shows a chevron at tintenthe UI.                   |
```
I feel like showArrow or something like that is a better name for this prop

youknowriad commented on packages/components/src/summary-button/README.md:
```diff
@@ -0,0 +1,42 @@
+| description       | string                    | -       | Optional supporting text that provides additional context or detail about the linked page.                                                                            |
+| strapline         | string                    | -       | A brief, optional line of text used to highlight important information, such as a warning or status.                                                                  |
+| decoration        | React.ReactElement        | -       | An optional visual element such as an icon or small illustration to enhance visual context or reinforce the category.                                                 |
+| fields            | SummaryButtonFieldProps[] | -       | This property is used to display `CoreBadge` instances per field. For this reason we need to define the props that match the `CoreBadge` component (intent and text). |
```
Should this be name "badges" rather than "fields"?

youknowriad commented on packages/components/src/core-badge/index.tsx:
```diff
@@ -0,0 +1,22 @@
+import { privateApis } from '@wordpress/components';
```
This is a duplication of an existing calypso component right? Should we extract this to its own PR and update the existing usage of the current component to avoid the unnecessary duplication in the code base?

ntsekouras replied:
Yeah, I mention this in the description. I was planning on doing it in a follow up, but can do it before.

ntsekouras replied:
https://github.com/Automattic/wp-calypso/pull/103069

ntsekouras committed:
rename props

ntsekouras committed:
remove types export

ntsekouras committed:
polish SummaryButton composing elements

ntsekouras committed:
update `medium` density styles

youknowriad commented:
This is looking good to me. Maybe mention the two new components in the changelog. and whether the CoreBadge is a follow-up or within this PR is your call.

ntsekouras committed:
add changelog entry and storybook `image` option for decoration

ntsekouras committed:
add some basic unit tests

ntsekouras committed:
fix docs

ntsekouras commented:
@jameskoster can you take a look when you find some time to see if we need to polish anything else - especially from your [comment](https://github.com/Automattic/wp-calypso/pull/102827#issuecomment-2827162282).

If there are just small css changes, feel free to push here. Thank you!

jameskoster commented:
<!-- linear:replyTo:2790604995 -->

@ntsekouras where do I test in storybook now the component moved?

ntsekouras commented:
>@ntsekouras where do I test in storybook now the component moved?

>yarn components:storybook:start

I updated the PR's description when I moved it.

ntsekouras committed:
polish from Jay

matt-west commented:
This is looking good. Thanks for shepherding it through @jameskoster.

mirka commented on packages/components/src/summary-button/index.stories.tsx:
```diff
@@ -0,0 +1,83 @@
+	args: {
+		title: 'Domain Settings',
+		description: 'Manage your domain settings, DNS, email, and more.',
+		onClick: () => alert( 'Clicked!' ),
```
For event listeners like these we usually just use a [Storybook action](https://storybook.js.org/docs/essentials/actions#via-storybooktest-fn-spy-function) for things like these, to be logged in the Actions panel.

mirka commented on packages/components/src/summary-button/README.md:
```diff
@@ -0,0 +1,42 @@
+# SummaryButton
```
For new components, can we omit manual READMEs for now? We can set up a README auto-generation step like we started to do in `@wordpress/components` if necessary, but without it, it just adds two sources of truth which is a hassle to maintain and eventually consolidate. It would be great if we could just keep TSDocs as the source.

ntsekouras replied:
Sounds good. Is there a plan though to add the auto generateed docs?

mirka replied:
On one hand I think a README isn't really necessary when you have IntelliSense docs and it also links out to a comprehensive docs site, but then again it might be useful for AI agents. So yes, pretty likely that we will add it.

ntsekouras committed:
feedback

ntsekouras commented:
I believe this is good to land as v1. I'd appreciate another review.

youknowriad commented on packages/components/src/summary-button/test/index.tsx:
```diff
@@ -0,0 +1,78 @@
+import SummaryButton from '../index';
+import { SummaryButtonBadgeProps } from '../types';
+
+describe( 'SummaryButton', () => {
```
Personally, I find this kind of tests useless, I would actually remove them most of the time that I update the component.

ntsekouras replied:
I agree as well. I added them at first for consistency with the others, but probably we should do it the other way around and remove them. I removed them from here.

ciampo replied:
While I agree that most of these tests were poorly written (for example, by getting elements by tag name instead of performing semantic queries), I still believe that unit tests would be useful, especially in terms of preventing future regressions on the most important behaviours of the component.

youknowriad replied:
@ciampo This is just a UI component without really behavior, what kind of unit tests do you see here?

ciampo replied:
Here are a few tests about behavior that come to mind:

- by default, it renders a `button` element (`button` role)
- when `href` is specified, it renders an `a` element (`link` role)
- `decoration` render prop is always rendered
- `strapline` prop is only rendered when `hasLowDensity`
- `description` prop is only rendered when `hasLowDensity`

Pretty much the list of tests that was deleted — IMO they are testing the behaviour and the consequences on the end user. Although we could make some improvements:

- instead of getting elements by tag name, query by role when possible (and by accessible text otherwise)
- use `@testing-library/user-event` instead of `fireEvent`
- use `toBeVisible` instead of `toBeInTheDocument` when possible

ntsekouras committed:
remove unit tests

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17428806

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @ntsekouras for including a screenshot in the description! This is really helpful for our translators.

ciampo commented on packages/components/src/summary-button/types.ts:
```diff
@@ -0,0 +1,74 @@
+	 */
+	text: string;
+	/**
+	 * Optional property to specify the color of the badge.
```
I wouldn't mention "color" — the fact that `Badge` has different colors based on the intent is an abstraction of `Badge`. I think that it would be more appropriate to say "specify the intent of the badge".

ntsekouras replied:
Good catch, thanks!

ciampo commented on packages/components/src/summary-button/types.ts:
```diff
@@ -0,0 +1,74 @@
+	 * @default true
+	 */
+	showArrow?: boolean;
+	/**
+	 * Determines if the element is disabled. If `true`, this will force a `button`
+	 * element to be rendered, even when an `href` is given.
+	 * @default false
+	 */
+	disabled?: boolean;
```
Switching from `a` to `button` is something that we should avoid — although this seems to be a behaviour coming from the core `Button` component.

Between the many style overrides and this sub-optimal disabled behaviour,  I wonder if we should just use a vanilla `button`.

Or if we should seriously think about a new a8c `Button` component that represents a better, simpler abstraction. (@mirka )

ntsekouras replied:
I actually tried to have a vanilla `button` but it didn't seem to worth it, as it required much of the Button code and styles (way more than the overrides we have).

ciampo reviewed (commented):
Even if this PR already got merged, I'm going to list a few aspects that IMO need improving / iterating:

- switching from `a` to `button` when `disabled` ([comment](https://github.com/Automattic/wp-calypso/pull/102827#discussion_r2066124405))
- improving style encapsulation around `decoration` ([comment](https://github.com/Automattic/wp-calypso/pull/102827#discussion_r2059609766))
- wording of around the badge intent ([comment](https://github.com/Automattic/wp-calypso/pull/102827#discussion_r2066101125))
- adding unit tests ([comment](https://github.com/Automattic/wp-calypso/pull/102827#discussion_r2066132603))


Also, what's the plan with the copy of the `Badge` component?

ciampo commented on packages/components/src/summary-button/types.ts:
```diff
@@ -0,0 +1,74 @@
+	 */
+	badges?: SummaryButtonBadgeProps[];
+	/**
+	 * A flag that indicates whether to show a chevron at the right-top UI.
```
Let's avoid "right" and "left" as they don't translate well for RTL languages
